terraform-organize-configuration-workshop/step5.md

```diff
@@ -0,0 +1,175 @@
+}
+```{{copy}}
+
+Notice that you did not configure a provider for this module. Modules inherit
+the provider configuration from the Terraform configuration that uses them.
+
+Like any Terraform configuration, modules can have variables and outputs.
+
+Add the following to `modules/aws-s3-static-website-bucket/variables.tf`{{open}}:
+
+```
+variable "bucket_name" {
+  description = "Name of the s3 bucket. Must be unique."
+  type        = string
+}
+```{{copy}}
+
+And add the following to `modules/aws-s3-static-website-bucket/outputs.tf`{{open}}:
+
+```
+output "arn" {
+  description = "ARN of the bucket"
+  value       = aws_s3_bucket.s3_bucket.arn
+}
+
+output "name" {
+  description = "Name (id) of the bucket"
+  value       = aws_s3_bucket.s3_bucket.id
+}
+
+output "website_endpoint" {
+  description = "Domain name of the bucket"
+  value       = aws_s3_bucket.s3_bucket.website_endpoint
+}
+```{{copy}}
+
+## Refactor Configuration
+
+Now refactor your `prod` and `dev` configuration to use this module.
+
+Update `dev/main.tf`{{open}} to remove the entire `resource "aws_s3_bucket"
+"web" { ... }` block, and replace it with the following:
+
+```
+module "website_s3_bucket" {
+  source = "../modules/aws-s3-static-website-bucket"
+
+  bucket_name = "${var.prefix}-${random_pet.petname.id}"
+}
+```{{copy}}
+
+And replace the bucket object resource block in the same file with the following
+to reference the bucket created by the module.
+
+```
+resource "aws_s3_bucket_object" "webapp" {
+  acl          = "public-read"
+  key          = "index.html"
+  bucket       = module.website_s3_bucket.name
+  content      = file("${path.module}/assets/index.html")
+  content_type = "text/html"
+}
+```{{copy}}
+
+You will also need to update `dev/outputs.tf`{{open}} to refer to the module instead of
+the resource name:
+
+```
+output "website_endpoint" {
+  description = "Website endpoint for this environment"
+  value       = "http://${module.website_s3_bucket.website_endpoint}/index.html"
+}
+```{{copy}}
+
+## Initialize and Apply
+
+Change into the dev directory and re-initialize it.
+
+```
+cd dev/
+terraform init
+```{{execute}}
+
+**Note**: Whenever you add or update a module, you will need to re-run
+`terraform init` or `terraform get` to install it, even if it is located on the
+same filesystem as your Terraform configuration.
+
+Now you can provision the bucket:
+
+```
+terraform apply -var-file=dev.tfvars
+```{{execute}}
+
+Respond `yes`{{execute}} to the prompt, and once again visit the website
+endpoint in your web browser to verify the website was deployed correctly.
+
+## Practice
+
+Now do the same with your prod environment.
+
+The steps to update and apply your production configuration are nearly identical
+to the ones for your dev environment. Be sure to apply your configuration with
+the `-var-file=prod.tfvars` flag.
+
+## Destroy Resources
+
+Clean up both environments by running `terraform destroy` in both directories.
```

Comment:
```suggestion
Clean up both environments by running `terraform destroy`{{execute}} in both directories.
```
These should be listed as clickable commands for both directories. Destruction is important, especially if they are using our AWS account.